### DISCUSSION TITLE
Navigator using ActionFormData

### DISCUSSION THREAD

**Discussion by RaphyTwin** [JaylyDev/ScriptAPI#134] [Q&A] [answered]
I don't know how I can make a teleport if I press a button. I sea the button but when I press it, nothing happen. My idea is it to get teleported to "0 64 0 facing 1 64 0". My code looks like this:

```
world.events.beforeItemUse.subscribe(event => {
   if (event.item.typeId === "minecraft:compass" && event.item.nameTag === "Navigator") {
      let form = new ActionFormData();
      form.title("Navigator");
      form.button("Hub", "textures/items/compass_item");

   form.show(event.source).then(r => {
      if (r.canceled) return;

      let response = r.selection;
      switch (response) {
         case 0:
            // Code to get teleported
            break;
         default:
      }
   }).catch(e => {
   console.error(e, e.stack);
});
   };
});
```
Thanks to all took time to read it and helps me.
(Sorry if my English is bad)

**Comment by JaylyDev** [comment]
you can use the [`.teleportFacing()`](https://learn.microsoft.com/en-us/minecraft/creator/scriptapi/minecraft/server/player#teleportfacing) function on player, which in this case you get it from `event.source`.

**Reply by RaphyTwin** [reply]
Thanks for replying, is it possible to give me a code example?

**Reply by JaylyDev** [reply]
```js
player.teleportFacing(
  { x: 0, y: 1, z: 0 },             // location
  world.getDimension('overworld'),  // dimension
  { x: 100, y: 100, z: 100 }        // facing location
)
```

**Reply by RaphyTwin** [reply]
Thanks. There is a error message saying: [Scripting][error]-ReferenceError 'player' is not defined   at <anonymous> (index.js:76)

**Reply by bot174** [reply, marked as answer]
player is an instance of the Player class in Jayly's code. In your code event.source returns an instance of the Player class, which you should do is `event.source.teleportFacing()`

**Reply by RaphyTwin** [reply]
Thanks. Now it says: [Scripting][error]-ReferenceError: Native onkect bound to prototype does not exist.   at <anonymus> (index.js:86)
